### PR TITLE
Update 0.14 - dependencies and build failure fixes

### PR DESCRIPTION
**Pull request by mvz**
## Summary

Update childprocess dependency and fix build failures for the 0.14 series

## Details

* Updates childprocess dependency
* Cherry pick several fixes from master
* Fix some 0.14-specific build failures

## Motivation and Context

It's annoying that the released version of Aruba pulls in an outdated runtime dependency. Since 1.0.0 will take a little while to finish, it's good to keep 0.14.x in shape. Also, there were some build failures so these have been fixed as well.

## How Has This Been Tested?

Travis 🙂 

## Types of changes

- Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by maxmeyer** [comment]
@mvz This is work in progress?

**Comment by mvz** [comment]
@maxmeyer yes, definitely :-)

File: lib/aruba/processes/spawn_process.rb [hunk 1]
```diff
@@ -92,7 +95,7 @@ def start
       end
       # rubocop:disable Metrics/MethodLength
 
-      # Access to stdout of process
+      # Access to stdin of process
```
Comment (olleolleolle) [inline lib/aruba/processes/spawn_process.rb]: Great find!

**Comment by maxmeyer** [comment]
@mvz Ok. I flagged this as work in progress.

**Comment by maxmeyer** [comment]
Wow, that looks like a big amout of work. Thanks that you work on this one!

**Comment by mvz** [comment]
This is ready for review.

**Comment by maxmeyer** [comment]
LGTM. Merged.

**Pull request merged**